In [37]:
# Read in the data 
import csv
with open('us-states.csv') as file:
        state_covid_data = []
        for row in csv.reader(file):
            state_covid_data.append(row)

In [38]:
# Look at the number of observations
len(state_covid_data)


10080

In [39]:
# Print the first five observations
state_covid_data[:5]

[['date', 'state', 'fips', 'cases', 'deaths'],
 ['2020-01-21', 'Washington', '53', '1', '0'],
 ['2020-01-22', 'Washington', '53', '1', '0'],
 ['2020-01-23', 'Washington', '53', '1', '0'],
 ['2020-01-24', 'Illinois', '17', '1', '0']]

### Count the number of unique dates in the data

In [43]:
# Create a set to store only the dates
dates = set()

# Loop through the list to put all the dates into the set made above
for observations in state_covid_data[1:]:
    dates.add(row[0])

# Check the length of dates
len(dates)

225

There are 225 unique dates in the data

### Find the first date in which the District of Columbia recorded a case. 

In [55]:
# Make a list with all the state names
states = []

# Loop through the data and put all the states into the above list
for observation in state_covid_data[1:]:
    states.append(observation[1])

# use index to find the first position of District of Columbia
states.index("District of Columbia")

#Since the index is 353 use 354 (to account for the first row with the variable names) and index 0 to see the date 
state_covid_data[354][0]

'2020-03-07'

The first date DC appears is 2020-03-07

### Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date of its first case.

In [72]:
def firstcoviddate():
  state = input ("enter state name")
  state_index=states.index(state)
  first_date=state_covid_data[state_index+1][0]
  return first_date

firstcoviddate()

ValueError: 'escape' is not in list

### Bonus: Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date when the number of reported cases within the state exceeded 1000.